khal/khalendar/event.py

```diff
@@ -132,143 +208,124 @@ def symbol_strings(self):
+        tzs = list()
+        for vevent in self._vevents.values():
+            if hasattr(vevent['DTSTART'].dt, 'tzinfo') and vevent['DTSTART'].dt.tzinfo is not None:
+                tzs.append(vevent['DTSTART'].dt.tzinfo)
+            if 'DTEND' in vevent and hasattr(vevent['DTEND'].dt, 'tzinfo') and \
+                    vevent['DTEND'].dt.tzinfo is not None and \
+                    vevent['DTEND'].dt.tzinfo not in tzs:
+                tzs.append(vevent['DTEND'].dt.tzinfo)
+
+        for tzinfo in tzs:
+            timezone = create_timezone(tzinfo, self.start)
+            calendar.add_component(timezone)
+
+        for vevent in self._vevents.values():
+            calendar.add_component(vevent)
+        return calendar.to_ical().decode('utf-8')
 
-        :rtype: str
-        :returns: event description
-        """
-        if self.allday:
-            end = self.end - timedelta(days=1)
-            if self.start == end:
-                rangestr = self.start.strftime(self.locale['longdateformat'])
-            else:
-                if self.start.year == self.end.year:
-                    startstr = self.start.strftime(self.locale['dateformat'])
-                else:
-                    startstr = self.start.strftime(self.locale['longdateformat'])
-                endstr = end.strftime(self.locale['longdateformat'])
-                rangestr = startstr + ' - ' + endstr
-        else:
-            # same day
-            if self.start.utctimetuple()[:3] == self.end.utctimetuple()[:3]:
-                starttime = self.start.strftime(self.locale['timeformat'])
-                endtime = self.end.strftime(self.locale['timeformat'])
-                datestr = self.end.strftime(self.locale['longdateformat'])
-                rangestr = starttime + '-' + endtime + ' ' + datestr
-            else:
-                startstr = self.start.strftime(self.locale['longdatetimeformat'])
-                endstr = self.end.strftime(self.locale['longdatetimeformat'])
-                rangestr = startstr + ' - ' + endstr
-            if self.start.tzinfo.zone != self.locale['local_timezone'].zone:
-                # doesn't work yet
-                # TODO FIXME
-                pass
-
-        location = '\nLocation: ' + self.location if \
-            self.location is not None else ''
-        description = '\nDescription: ' + self.description if \
-            self.description is not None else ''
-        repitition = '\nRepeat: ' + to_unicode(self.recurpattern) if \
-            self.recurpattern is not None else ''
+    @property
+    def ident(self):
+        """neeeded for vdirsyncer compat"""
+        return self._vevents[self.ref]['UID']
 
-        return '{}: {}{}{}{}'.format(
-            rangestr, self.summary, location, repitition, description)
+    @property
+    def summary(self):
+        return self._vevents[self.ref].get('SUMMARY', '')
+
+    def update_summary(self, summary):
```

Comment:
while this might make sense at the moment, all those update methods will soon get a `ref` and `thisandfuture` flag.